pkg/storage/bloom/v1/builder.go

```diff
@@ -64,42 +68,105 @@ type SeriesWithBloom struct {
 	Bloom  *Bloom
 }
 
-func (b *BlockBuilder) BuildFrom(itr Iterator[SeriesWithBloom]) (uint32, error) {
-	for itr.Next() {
-		if err := b.AddSeries(itr.At()); err != nil {
-			return 0, err
+type updatableBounds SeriesBounds
+
+func newUpdatableBounds() updatableBounds {
+	return updatableBounds{
+		FromTs:    model.Latest,
+		ThroughTs: model.Earliest,
+		FromFp:    model.Fingerprint(math.MaxInt64),
+		ThroughFp: model.Fingerprint(0),
+	}
+}
+
+func (b *updatableBounds) update(series *Series) {
+	minFrom := model.Latest
+	maxThrough := model.Earliest
+	for _, chunk := range series.Chunks {
+		if minFrom > chunk.Start {
+			minFrom = chunk.Start
+		}
+		if maxThrough < chunk.End {
+			maxThrough = chunk.End
 		}
+	}
 
+	if b.FromTs.After(minFrom) {
+		b.FromTs = minFrom
+	}
+	if b.ThroughTs.Before(maxThrough) {
+		b.ThroughTs = maxThrough
+	}
+	if b.FromFp > series.Fingerprint {
+		b.FromFp = series.Fingerprint
+	}
+	if b.ThroughFp < series.Fingerprint {
+		b.ThroughFp = series.Fingerprint
+	}
+}
+
+func (b *BlockBuilder) BuildFrom(itr Iterator[SeriesWithBloom]) (uint32, SeriesBounds, error) {
+	bounds := newUpdatableBounds()
+
+	for itr.Next() {
+		series := itr.At()
+		bounds.update(series.Series)
+		blockFull, err := b.AddSeries(series)
+		if err != nil {
+			return 0, SeriesBounds{}, err
+		}
+		if blockFull {
+			break
+		}
 	}
 
 	if err := itr.Err(); err != nil {
-		return 0, errors.Wrap(err, "iterating series with blooms")
+		return 0, SeriesBounds{}, errors.Wrap(err, "iterating series with blooms")
 	}
 
 	checksum, err := b.blooms.Close()
 	if err != nil {
-		return 0, errors.Wrap(err, "closing bloom file")
+		return 0, SeriesBounds{}, errors.Wrap(err, "closing bloom file")
 	}
 	if err := b.index.Close(); err != nil {
-		return 0, errors.Wrap(err, "closing series file")
+		return 0, SeriesBounds{}, errors.Wrap(err, "closing series file")
+	}
+	return checksum, SeriesBounds(bounds), nil
+}
+
+func (b *BlockBuilder) blockIsFull() (bool, error) {
```

Comment:
I think idiomatic Go suggests using `is`/`has` prefix for predicate functions

```suggestion
func (b *BlockBuilder) isBlockFull() (bool, error) {
```

Reply:
Done, thank you!